src/features/user/Profile/components/RedeemModal.tsx

```diff
@@ -74,23 +77,18 @@ export default function RedeemModal({
   async function validateCode(data: any) {
     setIsUploadingData(true);
     const submitData = {
-      type: 'gift',
+      // type: 'gift',
       code: data.code,
     };
     if (contextLoaded && user) {
-      const userLang = localStorage.getItem('language') || 'en';
-      postAuthenticatedRequest(
-        `/api/v1.3/${userLang}/validateCode`,
-        submitData,
-        token
-      ).then((res) => {
-        if (res.code === 401) {
-          setErrorMessage(res.message);
+      postAuthenticatedRequest(`/app/redeem`, submitData, token).then((res) => {
+        if (res.error_code === 'already_redeemed') {
+          setErrorMessage(t('me:alreadyRedeemed'));
           setIsUploadingData(false);
-        } else if (res.status === 'error') {
-          setErrorMessage(res.errorText || t('me:wentWrong'));
+        } else if (res.error_code === 'invalid_code') {
+          setErrorMessage(t('me:invalidCode'));
           setIsUploadingData(false);
```

Comment:
1. Both errors have similar handling, so you could combine them into one block. 
2. What happens if an unanticipated error is returned? It seems to me that the app will not inform the user about the error in that case.